app/web_hub.go

```diff
@@ -387,42 +419,91 @@ func (h *Hub) Start() {
 	doStart = func() {
 		mlog.Debug("Hub is starting", mlog.Int("index", h.connectionIndex))
 
-		connIndex := newHubConnectionIndex()
+		ticker := time.NewTicker(inactiveConnReaperInterval)
+		defer ticker.Stop()
+
+		connIndex := newHubConnectionIndex(inactiveConnReaperInterval)
 
 		for {
 			select {
 			case webSessionMessage := <-h.checkRegistered:
 				conns := connIndex.ForUser(webSessionMessage.userID)
 				var isRegistered bool
 				for _, conn := range conns {
+					if !conn.active {
```

Comment:
Do all these operations (read/write) on `WebConn.active` need to be atomic?

Reply:
Nope. I have checked this with `-race` flag. The `active` bit is only set during initialization. And it's always read/written in the hub loop, which always runs as a single goroutine. Didn't want to unnecessarily bring more atomics.

Reply:
Fair enough. Let's just be careful with this because now we have a mixed (getters/setters and direct) way of accessing connection fields based on where in the code we use them so it can get tricky and error prone.

Reply:
Fair point. I am assuming this part of the codebase is so sensitive, very few people are going to touch this. But I'll add a comment to the field for good measure.